src/Module/Race.cpp

```diff
@@ -6,6 +6,90 @@
 
 #include <Vehicle/TireControlE.h>
 
+void Race::Initialize(VehiclePtr vehicle) {
+    sf::Vector2f pos = map_->GetStartPosition();
+    vehicle_ = vehicle;
+    // Prepare map
+    vehicle_->Initialize(world_, (int) pos.x, (int) pos.y);
+
+    checkPointManager_->Reset();
+    hud_->Initialize(vehicle_, checkPointManager_);
+    map_->AlignCameraViewSize(*window_);
+    map_->SetCameraViewPosition(vehicle_->GetPosition());
+    world_->SetContactListener(&contactListener_);
```

Comment:
Dla spójności moglibyśmy przechowywać contactListener jako smart ptr w klasie. To samo worldem box2d.